sway/xdg_decoration.c

```diff
@@ -23,8 +23,34 @@ static void xdg_decoration_handle_request_mode(struct wl_listener *listener,
 		void *data) {
 	struct sway_xdg_decoration *deco =
 		wl_container_of(listener, deco, request_mode);
+	struct sway_view *view = deco->view;
+	enum wlr_xdg_toplevel_decoration_v1_mode mode =
+		WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE;
+	enum wlr_xdg_toplevel_decoration_v1_mode client_mode =
+		deco->wlr_xdg_decoration->client_pending_mode;
+
+	bool floating;
+	if (view->container) {
+		floating = container_is_floating(view->container);
+		bool csd = false;
+		if (floating) {
+			csd = client_mode ==
```

Comment:
Since `view_update_csd_from_client` already has a `container_is_floating` check, this `if` is unnecessary, we can set `csd` to the client's preference unconditionally.